src/synth/fluid_synth.c

```diff
@@ -3704,45 +3714,84 @@ fluid_synth_process_LOCAL(fluid_synth_t *synth, int len, int nfx, float *fx[],
     float cpu_load;
 
     fluid_return_val_if_fail(synth != NULL, FLUID_FAILED);
+
+    /* fx NULL while nfx > 0 is invalid */
+    fluid_return_val_if_fail((fx != NULL) || (nfx == 0), FLUID_FAILED);
+    /* nfx must be multiple of 2. Note that 0 value is valid and
+       allows to skip mixing in fx output buffers
+    */
     fluid_return_val_if_fail(nfx % 2 == 0, FLUID_FAILED);
+
+    /* out NULL while nout > 0 is invalid */
+    fluid_return_val_if_fail((fx != NULL) || (nfx == 0), FLUID_FAILED);
+    /* nout must be multiple of 2. Note that 0 value is valid and
+       allows to skip mixing in out output buffers
+    */
     fluid_return_val_if_fail(nout % 2 == 0, FLUID_FAILED);
+
+    /* check len value. Note than 0 value is valid, the function does nothing and
+       return FLUID_OK.
+    */
     fluid_return_val_if_fail(len >= 0, FLUID_FAILED);
     fluid_return_val_if_fail(len != 0, FLUID_OK); // to avoid raising FE_DIVBYZERO below
 
+    /*
+      Now len is > 0. This will lead in rendering in internal mixer buffer.
+      However this rendering must not occur if both nfx and nout are set to 0.
+      So now we check that both nfx and nout set to 0 while len > 0, is invalid.
+    */
+    fluid_return_val_if_fail(nfx || nout, FLUID_FAILED);
```

Comment:
This change is incorrect and breaks the unit test. If `len>0` while all buffers are NULL, rvoice_mixer will still render `len `samples, but nothing will be mixed.

Reply:
Ok.